### ISSUE TITLE
Completions encoding has regressed

### ISSUE THREAD

**Issue by lukechurch** [dart-lang/dart-services#312]
Response before refactoring:
```
{
 "replacementOffset": 284,
 "replacementLength": 0,
 "completions": [
  {
   "kind": "INVOCATION",
   "relevance": "1198",
   "completion": "toString",
   "selectionOffset": "8",
   "selectionLength": "0",
   "isDeprecated": "false",
   "isPotential": "false",
   "docSummary": "Returns a String-representation of this integer.",
   "docComplete": "Returns a String-representation of this integer.\n\nThe returned string is parsable by [parse].\nFor any `int` [:i:], it is guaranteed that\n[:i == int.parse(i.toString()):].",
   "declaringType": "int",
   "element": "{\"kind\":\"METHOD\",\"name\":\"toString\",\"location\":{\"file\":\"/usr/lib/dart/lib/core/int.dart\",\"offset\":8437,\"length\":8,\"startLine\":271,\"startColumn\":10},\"flags\":1,\"parameters\":\"()\",\"returnType\":\"String\"}",
   "returnType": "String",
   "parameterNames": "[]",
   "parameterTypes": "[]",
   "requiredParameterCount": "0",
   "hasNamedParameters": "false"
  },
...
```

After:
```
{
 "replacementOffset": 284,
 "replacementLength": 0,
 "completions": [
  {
   "kind": "INVOCATION",
   "relevance": "1198",
   "completion": "toString",
   "selectionOffset": "8",
   "selectionLength": "0",
   "isDeprecated": "false",
   "isPotential": "false",
   "docSummary": "Returns a String-representation of this integer.",
   "docComplete": "Returns a String-representation of this integer.\n\nThe returned string is parsable by [parse].\nFor any `int` [:i:], it is guaranteed that\n[:i == int.parse(i.toString()):].",
   "declaringType": "int",
   "element": "[Element kind: METHOD, name: toString, flags: 1]",
   "returnType": "String",
   "parameterNames": "[]",
   "parameterTypes": "[]",
   "requiredParameterCount": "0",
   "hasNamedParameters": "false"
  },
```

See the difference in the encoding of 'element'.

**Comment by lukechurch** [comment]
This is the result of just calling toString on an element, rather than doing a JSON encoding of it

**Comment by lukechurch** [comment]
Tentative fix: https://github.com/devoncarew/analysis_server_lib/pull/9

**Comment by lukechurch** [comment]
Fixed via version 0.1.2+1 of analysis_server_lib
